contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -98,15 +100,60 @@ def resolve_target(self, node_task, target, results_dir, node_paths, resolve_loc
         if not os.path.exists('yarn.lock') and frozen_lockfile:
           raise TaskError(
             'Cannot find yarn.lock. Did you forget to put it in target sources?')
-
-      result, command = node_task.install_module(
-        target=target, install_optional=install_optional,
-        production_only=production_only, force=force, frozen_lockfile=frozen_lockfile,
-        workunit_name=target.address.reference(),
-        workunit_labels=[WorkUnitLabel.COMPILER])
-      if result != 0:
-        raise TaskError('Failed to resolve dependencies for {}:\n\t{} failed with exit code {}'
-                        .format(target.address.reference(), command, result))
+      # Install all dependencies except for `file:` dependencies
+      # `file:` dependencies are special dependencies that point to a local path to a pants node target
+      # `file:` dependencies are already in the build graph and should be already be installed by this point
+
+      # Copy the package.json and then remove the file: dependencies from package.json
+      # Run the install and symlink the file: dependencies using their node_paths
+      # Afterwards, restore the original package.json to not cause diff changes when resolve_locally=True
+      with safe_temp_edit('package.json') as package_json:
+        with open(package_json, 'r') as package_json_file:
+          json_data = json.load(package_json_file)
+          source_deps = { k : v for k,v in json_data.get('dependencies', {}).items() if v.startswith('file:')}
```

Comment:
This is the second location doing `file:` extraction. Would recommend centralizing the parsing code somewhere (and if it's possible to just strip `file:` rather than using a regex, that would almost certainly be faster.

Reply:
This would still be good to do.